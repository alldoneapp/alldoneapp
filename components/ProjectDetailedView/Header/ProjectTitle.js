import React from 'react'
import { StyleSheet, View, Text } from 'react-native'
import styles from '../../styles/global'

const ProjectTitle = ({ project }) => {
    return (
        <View style={{ height: 64, flex: 1 }}>
            <View style={localStyles.upperContainer} />
            <View style={localStyles.bottomContainer}>
                <Text style={[styles.title4]}>{project.name}</Text>
            </View>
        </View>
    )
}

export default ProjectTitle

const localStyles = StyleSheet.create({
    upperContainer: {
        height: 32,
        backgroundColor: 'white',
    },
    bottomContainer: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'flex-start',
        alignItems: 'center',
        backgroundColor: 'white',
        height: 32,
    },
})
