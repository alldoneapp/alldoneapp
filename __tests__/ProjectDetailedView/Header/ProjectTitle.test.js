/**
 * @jest-environment jsdom
 */

import React from 'react'
import ProjectTitle from '../../../components/ProjectDetailedView/Header/ProjectTitle'
import renderer from 'react-test-renderer'
import store from '../../../redux/store'
import { storeLoggedUserProjects } from '../../../redux/actions'

describe('Detailed Project Title component', () => {
    describe('Detailed Project Title snapshot test', () => {
        it('should render correctly', () => {
            store.dispatch(storeLoggedUserProjects([{ name: 'My Project' }]))
            const tree = renderer.create(<ProjectTitle projectIndex={0} />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
