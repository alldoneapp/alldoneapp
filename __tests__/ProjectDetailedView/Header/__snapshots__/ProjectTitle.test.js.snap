// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detailed Project Title component Detailed Project Title snapshot test should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 64,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": 32,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "flexDirection": "row",
        "height": 32,
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#04142F",
            "fontFamily": "Roboto-Medium",
            "fontSize": 24,
            "lineHeight": 32,
          },
        ]
      }
    >
      My Project
    </Text>
  </View>
</View>
`;
